core/types/transaction.go

```diff
@@ -228,12 +230,11 @@ func (tx *Transaction) GasPrice() *big.Int                   { return new(big.In
 func (tx *Transaction) Value() *big.Int                      { return new(big.Int).Set(tx.data.Amount) }
 func (tx *Transaction) Nonce() uint64                        { return tx.data.AccountNonce }
 func (tx *Transaction) CheckNonce() bool                     { return true }
+func (tx *Transaction) SetNonce(nonce uint64)                { tx.data.AccountNonce = nonce }
```

Comment:
This is useful for the tx ingestion signing key. The nonce needs to be fetched right before signing and this method is called. Since the `transaction` struct already differs from the base geth `transaction`, I think its ok to add this.